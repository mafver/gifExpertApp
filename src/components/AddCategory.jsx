import PropTypes from "prop-types";
import { useState } from "react"

export const AddCategory = ( { onNewCategory } ) => {
    
    const [inputValue, setinputValue] = useState('');

    const onInputChange = ( {target} ) =>{
        setinputValue( target.value );
    }

    const onSubmit = ( event ) =>{
        event.preventDefault();
        // Para evitar que se mande un car√°cter o espacio en blanco.
        if ( inputValue.trim().length <= 1) return;
        onNewCategory( inputValue.trim() );
        // Limpiar el texto del input
        setinputValue('');
    }

    return (
        <form onSubmit={ onSubmit } aria-label="form">
            <input 
                type="text" 
                placeholder="Buscar gifs"
                value={ inputValue }
                onChange={ (event)=> onInputChange(event) }
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}