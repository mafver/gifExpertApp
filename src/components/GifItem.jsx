export const GifItem = ({id,title, url}) => {
  return (
    <div className="card">
        <img src={ url } alt= { title } />
        <br />
        <p>{ title }</p>
    </div>
  )
}
