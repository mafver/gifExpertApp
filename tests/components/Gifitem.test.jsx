import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas en <Giftitem.jsx>', () => { 
    const url = "https://one-punch-man.com/saitama.com"
    const title = "one punch man"
    test('Debe de verificar el Snapshot', () => { 
        const { container }  = render( <GifItem url={ url } title={ title }/>);
        expect( container ).toMatchSnapshot();
    });

    test('should Debe de mostar la imagen con el URL y el ALT indicado', () => { 
        render( <GifItem title={ title } url={ url }/>);
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    });

    test('should Debe de mostrar el tÃ­tulo en el componente', () => { 
        render( <GifItem title={ title } url={ url }/>);
        expect( screen.getByText(title) ).toBeTruthy();
    })
});
        