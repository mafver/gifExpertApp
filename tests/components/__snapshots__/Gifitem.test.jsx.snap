// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Giftitem.jsx> Debe de verificar el Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="one punch man"
      src="https://one-punch-man.com/saitama.com"
    />
    <br />
    <p>
      one punch man
    </p>
  </div>
</div>
`;
